src/main/java/core/basesyntax/Engine.java

```diff
@@ -1,5 +1,21 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
+    private String type;
+    private int horsePower;
+    private int mileage;
+    private int weight;
+    private int age;
 
+    Engine(String type, int horsePower, int mileage, int weight, int age) {
+        this.type = type;
+        this.horsePower = horsePower;
+        this.mileage = mileage;
+        this.weight = weight;
+        this.age = age;
+    }
+
+    public Engine clone() {
+        return new Engine(type, horsePower, mileage, weight, age);
```

Comment:
@NikolayKatsai why you are not using new String(type) as in Car class?

Reply:
@NikolayKatsai check the both cases: new String(type) and type. Clone the object. Change the type param. Will you see any difference?

Reply:
Yep, sorry, forgot to fix it in Engine class))